LibraBFT/Impl/Consensus/Network.agda

```diff
@@ -9,10 +9,53 @@ open import LibraBFT.ImplShared.Base.Types
 open import LibraBFT.ImplShared.Consensus.Types
 open import LibraBFT.ImplShared.NetworkMsg
 open import LibraBFT.ImplShared.Util.Util
+open import LibraBFT.Impl.Consensus.ConsensusTypes.ProposalMsg as ProposalMsg
+import      LibraBFT.Impl.Consensus.ConsensusTypes.VoteMsg     as VoteMsg
 open import LibraBFT.Prelude
 open import Optics.All
 
 module LibraBFT.Impl.Consensus.Network where
 
-postulate  -- TODO-1: implement this
-  processProposal : {- NodeId → -} ProposalMsg → Epoch → ValidatorVerifier → (Either ErrLog InfoLog) ⊎ Unit
+processProposal : {- NodeId → -} ProposalMsg → Epoch → ValidatorVerifier → Either (Either ErrLog InfoLog) Unit
+processProposal {- peerId -} proposal myEpoch vv =
+  case pProposal of λ where
+    (Left e) → Left (Left e)
+    (Right unit) →
```

Comment:
Question: is this binding `unit` or pattern matching on `unit`?

Reply:
> Question: is this binding `unit` or pattern matching on `unit`?

This is a pattern match. Once you use `where`, you can do pattern matching arbitrarily deeply.

Reply:
magic `where` - I'll remember that.